src/Classes/StringUtil.php

```diff
@@ -127,4 +168,57 @@ public static function formatFileSize(int $bytes): string
 
         return $bytes;
     }
+
+    /**
+     * Clean the tinyMCE data, see rules below
+     * Rule #1 : Replace [nbsp] tags by ' '
+     * Rule #2 : Find special characters and add an [nbsp] just before.
+     *
+     * @param string $varValue [Value to clean]
+     */
+    public static function cleanSpaces(string $varValue): string
+    {
+        // Rule #1
+        $varValue = str_replace(['[nbsp]', '&nbsp;'], [' ', ' '], $varValue);
+
+        // Rule #2
+        $varValue = preg_replace("/\s(\?|\!|\:|\;|\»)/", '&nbsp;\\1', $varValue);
+
+        return preg_replace("/(\«)\s/", '\\1&nbsp;', $varValue);
+    }
+
+    /**
+     * Generates a random key of specified length.
+     *
+     * @param int|null $length The length of the key (default is 16)
+     * @return string The random key generated
+     * @throws \Exception
+     */
+    public static function generateKey(?int $length = 16): string
+    {
+        $characters = 'abcdefghijklmnopqrstuwxyzABCDEFGHIJKLMNOPQRSTUWXYZ0123456789_-';
+        $randstring = '';
+        for ($i = 0; $i < $length; ++$i) {
+            $randstring .= $characters[random_int(0, \strlen($characters) - 1)];
+        }
+
+        return $randstring;
+    }
+
+    public static function getFormStorageDataValueAsString($mixed): string
```

Comment:
@NodokaMurmevent Why is this here ? Shouldn't it belong to the https://github.com/Web-Ex-Machina/contao-form-data-manager bundle ?

Reply:
it’s used here [contao-form-data-manager UiListener.php line 212](https://github.com/Web-Ex-Machina/contao-form-data-manager/blob/e5b953f772bd990656a58c109dd117c863ce1b04/src/EventListener/PersonalDataManager/UiListener.php#L212)

but it's true, if the form data manager is the alone to use it, we can move it in this bundle.

Reply:
I reckon no one will call this function unless the FDM bundle is installed, in which case they could call it from there.
I'd prefer this function to be moved to the FDM bundle